lib/PuppeteerSharp.Tests/PublishSingleFileTests/PublishSingleFileTests.cs

```diff
@@ -0,0 +1,96 @@
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using PuppeteerSharp.Tests.Attributes;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.SingleFileDeployment
+{
+    public class PublishSingleFileTests
+    {
+        [SkipNonWindowsFact]
+        public void ShouldWork()
+        {
+            var tempPath = Path.GetTempPath();
+            var actualFilePath = Path.Combine(tempPath, $"google.jpg");
+            var actualWindowsBinary = DotnetPublishSingleFile("PuppeteerSharp.Tests.SingleFileDeployment");
+
+            DeleteIfExists(actualFilePath);
+
+            using var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = actualWindowsBinary,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    WorkingDirectory = tempPath
+                }
+            };
+
+            process.Start();
+            var outputResult = GetStreamOutput(process.StandardOutput);
+            var errorResult = GetStreamOutput(process.StandardError);
+            var isExited = process.WaitForExit(80000);
+
+            if (!isExited)
+            {
+                process.Kill();
+            }
+
+            Assert.True(isExited);
+
+            Assert.True(File.Exists(actualFilePath), $"StdOut: {outputResult}\nStdErr: {errorResult}\n");
+        }
+
+        private static string GetStreamOutput(StreamReader stream)
+        {
+            var outputReadTask = Task.Run(() => stream.ReadToEnd());
```

Comment:
Why do we need this `Task.Run`?

Reply:
I followed https://stackoverflow.com/questions/7160187/standardoutput-readtoend-hangs

Reply:
Does using [ReadToEndAsync](https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.readtoendasync?view=net-7.0) work around the issue?

Reply:
I will give it a try... I am not sure optimizing testcode is worth it... 😌

Reply:
I'll defer to @kblok on this one, what's your thoughts?

Reply:
As far as I remember reading StdOut and StdErr at the same time is causing deadlocks and using [ReadToEndAsync](https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.readtoendasync?view=net-7.0) wont fix that. The output of them isnt used to assert the expectation and could be dropped entirely but if the assertion will fail it is the information you will need to find out what went wrong.

Reply:
I've seen the following pattern used a few times, I've not used it myself yet.

https://stackoverflow.com/a/53504707/4583726